src/libxrpl/json/json_value.cpp

```diff
@@ -672,7 +672,8 @@ Value::isConvertibleTo(ValueType other) const
                 (other == intValue && value_.real_ >= minInt &&
                  value_.real_ <= maxInt) ||
                 (other == uintValue && value_.real_ >= 0 &&
-                 value_.real_ <= maxUInt) ||
+                 value_.real_ <= maxUInt &&
+                 static_cast<int>(value_.real_) == value_.real_) ||
```

Comment:
You should avoid using == with float point. Consider something like `std::fabs(round(value_.real_) - value_.real_) < 1e-10`

Reply:
We should use `std::numeric_limits<double>::epsilon()` instead of `1e-10`